template/java8/model/src/main/java/com/openfaas/model/Request.java

```diff
@@ -30,4 +42,42 @@ public String getHeader(String key) {
 
         return this.headers.get(key);
     }
+    
+    @Override
+	public String getQueryRaw() {
+		return queryRaw;
+	}
+
+    @Override
+	public Map<String, String> getQueryParameters() {
+    	return queryParameters;
+    }
+    
+	private Map<String, String> parseQueryParameters() {
+		Map<String, String> reqParametersMap = new HashMap<String, String>();
```

Comment:
Is there no utility method within Java to do this operation?

Reply:
Parsing can be done with some utility method but URLDecoder is needed to safely decode query parameters and this won't be supported in utility method.